File: sol/022.cpp
#include <iostream>
using namespace std;

long long gcd(long long a, long long b) {
	if (b == 0) return a;
	return gcd(b, a % b);
}

long long A, B, C;

int main() {
	cin >> A >> B >> C;
	long long S = gcd(A, gcd(B, C));
	cout << (A / S - 1LL) + (B / S - 1LL) + (C / S - 1LL) << endl;
	return 0;
}

File: sol/022.py
import math

# 入力を受け取る
a, b, c = map(int, input().split())

# math.gcd()で最大公約数を求める
g = math.gcd(math.gcd(a, b), c)

#答えを出力する
print((a+b+c)//g-3)
